File: 2021/js/day_1.js
const path = require('path');
const fs = require('fs');

const input = fs.readFileSync(path.join(__dirname, '..', 'inputs', 'day1.txt'), { encoding: 'utf-8' });
const measurements = input.split('\n').map(n => parseInt(n, 10));

function getNumberOfIncreases(measurements) {
  let counter = 0;
  for (i in measurements) {
    if (measurements[i] > measurements[i - 1]) {
      counter++;
    }
  }
  return counter
}

console.log(
  getNumberOfIncreases(measurements),
  'measurements are larger than the previous'
);

File: 2021/go/day_1/day_1.go
package day_1

import (
	"fmt"
	"strings"

	"2021/utils"
)

func getNumberOfIncreases(measurements []string) int {
	counter, intArray := 0, utils.ToIntArray(measurements)

	for i := 1; i < len(measurements); i++ {
		if intArray[i] > intArray[i - 1] {
			counter += 1
		}
	}

	return counter
}

func part1() {
	var measurements = strings.Split(utils.MyReadFile("day1.txt"), "\n")

	fmt.Printf(" - (Part 1) %d measurements are larger than the previous", getNumberOfIncreases(measurements))
}

func Solution() {
	fmt.Println("Day 1: Sonar Sweep")
	part1()
}